src/sql/workbench/services/objectExplorer/browser/treeUpdateUtils.ts

```diff
@@ -348,4 +357,83 @@ export class TreeUpdateUtils {
 		}
 		return connectionProfile;
 	}
+
+	/**
+	 * Change the connection title to display only the unique properties among profiles.
+	 */
+	private static alterConnectionTitles(inputList: ConnectionProfile[]): void {
+		let profileListMap = new Map<string, number[]>();
+
+		// Map the indices of profiles that share the same connection name.
+		for (let i = 0; i < inputList.length; i++) {
+			// do not add if the profile is still loading as that will result in erroneous entries.
+			if (inputList[i].hasServerCapabilities) {
+				let titleKey = inputList[i].getOriginalTitle();
+				if (profileListMap.has(titleKey)) {
+					let profilesForKey = profileListMap.get(titleKey);
+					profilesForKey.push(i);
+					profileListMap.set(titleKey, profilesForKey);
+				}
+				else {
+					profileListMap.set(titleKey, [i]);
+				}
+			}
+		}
+
+		profileListMap.forEach(function (value, key) {
+			if (profileListMap.get(key)?.length > 1) {
+				let combinedOptions = [];
+				let needSpecial = false;
+				if (key === inputList[value[0]].connectionName) {
+					// check for potential connections with the same name but technically different connections.
+					let listOfDuplicates = value.filter(item => inputList[item].getOptionsKey() !== inputList[value[0]].getOptionsKey());
+					if (listOfDuplicates.length > 0) {
+						// if we do find duplicates, we will need to include the special properties.
+						needSpecial = true;
+					}
+				}
+				value.forEach((value) => {
+					// Add all possible options across all profiles with the same title to an option list.
+					let valueOptions = inputList[value].getConnectionOptionsList(needSpecial, false);
+					combinedOptions = combinedOptions.concat(valueOptions.filter(item => combinedOptions.indexOf(item) < 0));
+				});
+
+				for (let p = 0; p < value.length; p++) {
+					let firstOption = true;
+					for (let i = 0; i < combinedOptions.length; i++) {
+						// See if the option is not default for the inputList profile or is.
+						if (inputList[value[p]].getConnectionOptionsList(needSpecial, true).indexOf(combinedOptions[i]) > -1) {
+							let isUnique = true;
+							let optionValue = inputList[value[p]].getOptionValue(combinedOptions[i].name);
+							for (let t = 0; t < value.length; t++) {
+								// Check to make sure that the option value for this profile is unique among the other values with the same title.
+								if (p !== t) {
+									if (inputList[value[t]].getOptionValue(combinedOptions[i].name) === optionValue) {
+										isUnique = false;
+										break;
+									}
+								}
+							}
+
+							if (isUnique) {
+								// Add the option to the title and the brace in front if this is the first option.
+								if (firstOption) {
+									firstOption = false;
+									inputList[value[p]].title += ' (';
+								}
+								inputList[value[p]].title += (combinedOptions[i].name + ':' + optionValue + '|');
```

Comment:
Instead of `|` (pipe) can we use `; ` (semi-colon with a leading space) and `=` as name-value separator? As that's more readable and relatable with connection string.

![image](https://user-images.githubusercontent.com/13396919/226690567-941bb54a-6c3a-4665-a3e2-fa643f6f0e93.png)